In [2]:
import pandas as pd
import numpy as np
import geopandas as gp
import shapely
from planet4 import io, markings

In [3]:
from coverage_utils import CoverageCalculator, get_obsids_for_region
from my_io import get_tilecoords, get_metadata, get_region_names

In [5]:
olddata = pd.read_csv("n_obj_df.csv")

In [19]:
olddata

Latitude       Ls  MY     Which  fraction  n_objects            obsid  \
0    -85.4074  214.785  29      both     0.115        933  ESP_012079_0945   
1    -85.4065  183.727  29      both     0.070       1971  ESP_011407_0945   
2    -85.4056  198.040  29      both     0.072       1525  ESP_011723_0945   
3    -85.4049  207.892  29      both     0.168        788  ESP_011934_0945   
4    -85.4067  235.282  29      both     0.111       1285  ESP_012501_0945   
5    -85.4117  224.964  29      both     0.087        350  ESP_012290_0945   
6    -85.4086  183.683  29      both     0.074        453  ESP_011406_0945   
7    -85.4074  214.785  29      fans     0.088        781  ESP_012079_0945   
8    -85.4065  183.727  29      fans     0.046       1568  ESP_011407_0945   
9    -85.4056  198.040  29      fans     0.047        939  ESP_011723_0945   
10   -85.4049  207.892  29      fans     0.126        678  ESP_011934_0945   
11   -85.4067  235.282  29      fans     0.097       1253  ESP_012501_0945   
12   -85.4117  224.964  29      fans     0.083        326  ESP_012290_0945   
13   -85.4086  183.683  29      fans     0.026        291  ESP_011406_0945   
14   -85.4074  214.785  29  blotches     0.037        331  ESP_012079_0945   
15   -85.4065  183.727  29  blotches     0.029        529  ESP_011407_0945   
16   -85.4056  198.040  29  blotches     0.032        808  ESP_011723_0945   
17   -85.4049  207.892  29  blotches     0.057        230  ESP_011934_0945   
18   -85.4067  235.282  29  blotches     0.021        131  ESP_012501_0945   
19   -85.4117  224.964  29  blotches     0.006         47  ESP_012290_0945   
20   -85.4086  183.683  29  blotches     0.056        256  ESP_011406_0945   
21   -85.4041  243.978  30      both     0.049       2256  ESP_021494_0945   
22   -85.4071  209.383  30      both     0.075        209  ESP_020782_0945   
23   -85.4086  206.269  30      both     0.062        220  ESP_020716_0945   
24   -85.4078  199.501  30      both     0.065        560  ESP_020571_0945   
25   -85.4105  186.879  30      both     0.080        445  ESP_020294_0945   
26   -85.4075  216.293  30      both     0.079        164  ESP_020927_0945   
27   -85.4056  184.557  30      both     0.054        481  ESP_020242_0945   
28   -85.4041  243.978  30      fans     0.032       1053  ESP_021494_0945   
29   -85.4071  209.383  30      fans     0.050        145  ESP_020782_0945   
..        ...      ...  ..       ...       ...        ...              ...   
624  -86.2597  214.426  30      fans     0.033         53  ESP_020888_0935   
625  -86.2599  197.699  30  blotches     0.025         97  ESP_020532_0935   
626  -86.2535  242.693  30  blotches     0.027       2392  ESP_021468_0935   
627  -86.2587  185.136  30  blotches     0.058        208  ESP_020255_0935   
628  -86.2751  285.423  30  blotches     0.001         27  ESP_022339_0935   
629  -86.2597  214.426  30  blotches     0.011         36  ESP_020888_0935   
630  -85.0057  207.279  29      both     0.056        216  ESP_011921_0950   
631  -85.0057  207.279  29      fans     0.055        201  ESP_011921_0950   
632  -85.0057  207.279  29  blotches     0.002         30  ESP_011921_0950   
633  -73.4875  223.218  29      both     0.014       2024  ESP_012254_1065   
634  -73.4787  254.344  29      both     0.031       2805  ESP_012887_1065   
635  -73.4875  223.218  29      fans     0.004        399  ESP_012254_1065   
636  -73.4787  254.344  29      fans     0.002         31  ESP_012887_1065   
637  -73.4875  223.218  29  blotches     0.011       1740  ESP_012254_1065   
638  -73.4787  254.344  29  blotches     0.030       2790  ESP_012887_1065   
639  -86.9844  212.826  29      both     0.019        141  ESP_012038_0930   
640  -86.9850  243.661  29      both     0.010        363  ESP_012671_0930   
641  -86.9846  250.830  29      both     0.007        400  ESP_012816_0930   
642  -86.9844  212.826  29      fans     0.018        100  ESP_012038_0930   
643  -86.9850  243.661  29 

In [7]:
data = pd.read_csv("P4_catalog_v1.1_tile_coords_final.csv")

In [31]:
data.head()

x_tile  y_tile  x_hirise  y_hirise  PlanetocentricLatitude  \
0       1       1     420.0     324.0              -85.504144   
1       1       2     420.0     872.0              -85.502300   
2       1       3     420.0    1420.0              -85.500454   
3       1       4     420.0    1968.0              -85.498607   
4       1       5     420.0    2516.0              -85.496762   

   PlanetographicLatitude  PositiveEast360Longitude  BodyFixedCoordinateX  \
0              -85.556701                104.952104            -68.380674   
1              -85.554878                104.934805            -68.331337   
2              -85.553054                104.917486            -68.281853   
3              -85.551229                104.900198            -68.232522   
4              -85.549404                104.882968            -68.183281   

   BodyFixedCoordinateY  BodyFixedCoordinateZ     tile_id            obsid  
0            256.056147          -3370.636698  APF0000cwz  ESP_012079_0945  
1            256.181661          -3370.628630  APF0000ck9  ESP_012079_0945  
2            256.307249          -3370.620295  APF0000cty  ESP_012079_0945  
3            256.433112          -3370.615690  APF0000ciy  ESP_012079_0945  
4            256.558509          -3370.606455  APF0000cwp  ESP_012079_0945

In [32]:
#x1 = data[data['x_tile'] == 1]

" Through all the coordinates for all tiles for an obsid into a Shapely object and let shapely define the boundary. This would be good to learn, b/c that’s exactly what shapely was made for and it is a standard GIS task to learn."

In [44]:
data[data['obsid'] == 'ESP_012079_0945'].x_tile.max()

5

In [22]:
allobs = data.obsid.unique() #give me every obsid

In [60]:

xmiba = []
xmaba = []
ymiba = []
ymaba = []

for obs in allobs:
    temp = data[data['obsid'] == obs] #look only at our obsid in question
    #Get xmins, xmaxes for the tiles
    xmi = temp.x_tile.min()
    xma = temp.x_tile.max()
    ymi = temp.y_tile.min()
    yma = temp.y_tile.max()
    xmiba.append(xmi)
    xmaba.append(xma)
    ymiba.append(ymi)
    ymaba.append(yma)
    
xmiba = np.asarray(xmiba)
ymiba = np.asarray(ymiba)
xmaba = np.asarray(xmaba)    
ymaba = np.asarray(ymaba)

Running and analyzing the above shows that x_tile has a min of 1, max of 5 and the y_tile has a min of 1, max of 145 across all our obsids.

In [89]:
results = 1

In [131]:
#similar to above, but now to create a dataframe which will help us get the corner values
bucket = []
for obs in allobs:
    d = {}
    temp = data[data['obsid'] == obs]
    d["x_tile_min"] = temp.x_tile.min()
    d["x_tile_max"] = temp.x_tile.max()
    d["y_tile_min"] = temp.y_tile.min()
    d["y_tile_max"] = temp.y_tile.max()
    d["obsid"] = obs
    d["top_left"] = (temp[(temp['x_tile'] == temp.x_tile.min()) & (temp["y_tile"] == temp.y_tile.min())].BodyFixedCoordinateX.values, temp[(temp['x_tile'] == temp.x_tile.min()) & (temp["y_tile"] == temp.y_tile.min())].BodyFixedCoordinateY.values)
    d["top_right"] = (temp[(temp['x_tile'] == temp.x_tile.min()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateX.values, temp[(temp['x_tile'] == temp.x_tile.min()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateY.values )
    d['bottom_left'] = (temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.min())].BodyFixedCoordinateX.values, temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.min())].BodyFixedCoordinateX.values)
    d['bottom_right'] = (temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateX.values, temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateY.values )
    bucket.append(d)
results = pd.DataFrame(bucket)


In [132]:
results.head() #looks weird, why are the bottom left x and bottom right x values different? curvature?  
#also what to do when there are nan values?

bottom_left  \
0    ([-67.69881540000002], [-67.69881540000002])   
1    ([108.67046979999999], [108.67046979999999])   
2                                  ([nan], [nan])   
3  ([-319.50996000000004], [-319.50996000000004])   
4    ([227.52294890000002], [227.52294890000002])   

                                     bottom_right            obsid  \
0          ([-62.388268200000006], [269.3496037])  ESP_012079_0945   
1           ([121.8381508], [477.67216169999995])  ESP_011697_0980   
2                                  ([nan], [nan])  ESP_021455_0935   
3  ([-328.12714460000007], [-328.13045330000006])  ESP_012008_0975   
4           ([239.92119440000002], [417.6651198])  ESP_012436_0980   

                           top_left  \
0    ([-68.3806735], [256.0561472])   
1    ([108.3633138], [457.7678756])   
2                    ([nan], [nan])   
3  ([-319.1787501], [-324.5326963])   
4     ([227.279471], [406.9997462])   

                                       top_right  x_tile_max  x_tile_min  \
0  ([-63.070566500000005], [269.61576610000003])           5           1   
1          ([121.5308617], [477.87006569999994])           2           1   
2                                 ([nan], [nan])           5           1   
3               ([-327.7960465], [-328.7914651])           5           1   
4                  ([239.6778595], [417.937003])           2           1   

   y_tile_max  y_tile_min  
0         109           1  
1          91           1  
2          63           1  
3          72           1  
4          63           1

In [123]:
results.x_coord_min[0]

array([-68.3806735])

In [119]:
temp[(temp['x_tile'] == temp.x_tile.max()) & (temp["y_tile"] == temp.y_tile.max())].BodyFixedCoordinateX.values

array([221.6780802])